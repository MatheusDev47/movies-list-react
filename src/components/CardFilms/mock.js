export const cardFilmsPropsMock = {
    img: 'img/img.png',
    title: 'title 01',
    overview: 'body 01'
}