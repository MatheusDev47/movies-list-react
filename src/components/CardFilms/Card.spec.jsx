import { render, screen } from '@testing-library/react';
import { CardFilms } from '.';
import { cardFilmsPropsMock } from './mock';

const props = cardFilmsPropsMock;

describe('<CardFilms />', () => {
  it('Should CardFilms is render correctly in the page', () => {
    render(<CardFilms {...props} />);

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute('src', props.img);
    expect(screen.getByRole('heading', { name: props.title })).toBeInTheDocument();
    expect(screen.getByText(props.overview)).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = render(<CardFilms {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
