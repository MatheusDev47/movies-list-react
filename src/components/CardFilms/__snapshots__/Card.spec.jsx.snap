// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardFilms /> Should match snapshot 1`] = `
<div
  class="card"
>
  <img
    alt="title 01"
    src="img/img.png"
  />
  <div
    class="film-content"
  >
    <h2>
      title 01
    </h2>
    <p>
      body 01
    </p>
  </div>
</div>
`;
